daemon/containerd/image_push.go

```diff
@@ -62,13 +63,17 @@ func (i *ImageService) PushImage(ctx context.Context, targetRef reference.Named,
 	store := i.client.ContentStore()
 
 	resolver, tracker := i.newResolverFromAuthConfig(ctx, authConfig)
-	progress := pushProgress{Tracker: tracker}
+	pp := pushProgress{Tracker: tracker}
 	jobsQueue := newJobs()
 	finishProgress := jobsQueue.showProgress(ctx, out, combinedProgress([]progressUpdater{
-		&progress,
+		&pp,
 		pullProgress{ShowExists: false, Store: store},
 	}))
-	defer finishProgress()
+	defer func() {
+		finishProgress()
+		tagged := targetRef.(reference.Tagged)
```

Comment:
I don't think `PushImage` will get called with a non-tagged reference, but just in case, can we have an `if ok` check?

Reply:
It's tagged for sure https://github.com/moby/moby/pull/46384/files#diff-beb8239a1725706c868633c368e1259236a327aaba2ffdf2cf39b991ad0fa0c6L38

Reply:
Oh no

Reply:
`image@sha256:...` would not pass this check. It's only the API that makes sure that the reference is not digested.